tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -992,7 +992,8 @@ SpirvInstruction *SpirvEmitter::doExpr(const Expr *expr,
     assert(curThis);
     result = curThis;
   } else if (isa<CXXConstructExpr>(expr)) {
-    result = curThis;
+    if (!hlsl::IsHLSLRayQueryType(expr->getType()))
```

Comment:
It may be worth adding a short comment about why this check is needed, as it's not obvious at all when just reading the code.

Reply:
Done